src/test/unit/app/controller/courts/CourtTypesController.ts

```diff
@@ -0,0 +1,189 @@
+
+
+  const courtTypes: CourtType[] = [
+    { id: 1, name:"Magistrates' Court", code: 123},
+    { id: 2, name:'County Court', code: 456},
+    { id: 3, name:'Crown Court', code: 789},
+    { id: 4, name:'Family Court', code: null}
+  ];
+
+  const courtTypeItems: CourtTypeItem[] = [
+    {value:"1,Magistrates' Court", text:"Magistrates' Court", checked: true, code:123},
+    {value:'2,County Court', text:'County Court', checked: true, code:456},
+    {value:'3,Crown Court', text:'Crown Court', checked:true, code:789},
+    {value:'4,Family Court', text:'Family Court', checked:true, code:null}
+
+  ];
+
+  const courtCourtTypes: CourtType[] =[
+    { id: 1, name:"Magistrates' Court", code: 123},
+    { id: 2, name:'County Court', code: 456},
+    { id: 3, name:'Crown Court', code: 789},
+    { id: 4, name:'Family Court', code: null}
+  ];
+
+
+  const controller = new CourtTypesController();
+
+  beforeEach(() => {
+    mockApi = {
+      getCourtTypes: async (): Promise<CourtType[]> => courtTypes,
+      updateCourtCourtTypes: async (): Promise<CourtType[]> => courtTypes,
+      getCourtCourtTypes: async (): Promise<CourtType[]> => courtCourtTypes
+    };
+  });
+
+
+  test('Should get court types view and render the page', async () => {
+    const req = mockRequest();
+    req.params = {
+      slug: 'southport-county-court'
+    };
+    req.scope.cradle.api = mockApi;
+    const res = mockResponse();
+
+    await controller.get(req, res);
+
+    const expectedResults: CourtTypePageData = {
+      updated: false,
+      errorMsg: '',
+      items: courtTypeItems
+    };
+
+    expect(res.render).toBeCalledWith('courts/tabs/typesContent', expectedResults);
+  });
+
+  test('Should post court court types if court types are valid', async () => {
+    const slug = 'southport-county-court';
+    const res = mockResponse();
+    const req = mockRequest();
+
+    const types: string[]= [
+      "1,Magistrates' Court",
+      '2,County Court',
+      '3,Crown Court',
+      '4,Family Court'
+    ];
+
+    req.body = {
+      'types': types,
+      'magistratesCourtCode' : '123',
+      'countyCourtCode' : '456',
+      'crownCourtCode': '789'
+
+    };
+    req.params = { slug: slug };
+    req.scope.cradle.api = mockApi;
+    req.scope.cradle.api.updateCourtCourtTypes = jest.fn().mockResolvedValue(res);
+
+    await controller.put(req, res);
+
+    // Should call API to save data
+    expect(mockApi.updateCourtCourtTypes).toBeCalledWith(slug, courtCourtTypes);
+  });
+
+
+  test('Should not post court types if no court types is selected', async() => {
+    const slug = 'another-county-court';
+    const res = mockResponse();
+    const req = mockRequest();
+
+    req.body = {
+    };
+
+    req.params = { slug: slug };
+    req.scope.cradle.api = mockApi;
+    req.scope.cradle.api.updateCourtCourtTypes = jest.fn().mockReturnValue(res);
+
+    await controller.put(req, res);
+
+    // Should not call API if opening times data is incomplete
```

Comment:
Just a typo here referring to opening times

Reply:
removed comments